### PR TITLE
[YUNIKORN-853] Update the value type of node attribute to support Any type

### PR DESCRIPTION
**Pull request by chenya-zhang**
### What is this PR for?

Node labels need to be passed into YuniKorn as node attributes for node sorting. The type of node labels is a map of string to string.

The current value type of node attribute is string and cannot support a map. It can be updated to google.protobug.any: https://developers.google.com/protocol-buffers/docs/proto3#any

### What type of PR is it?
* [ ] - Bug Fix
* [X] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?

https://issues.apache.org/jira/browse/YUNIKORN-853

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Review by yangwwei — COMMENTED** [review]
Looks good,  just a minor comment.
But note, this is an incompatible change. we need to update the core repo accordingly and then the shim to use this latest message type.

File: scheduler-interface-spec.md [hunk 1]
```diff
@@ -640,6 +641,7 @@ const (
 	ARCH                = "si/arch"
 	HostName            = "si/hostname"
 	RackName            = "si/rackname"
+       LabelName           = "si/labelname"
```
Comment (yangwwei) [inline scheduler-interface-spec.md:644]: this is the key name of the attribute, to be more self-explanatory, would it be better to rename this as `si/nodelabels`?

**Review by yangwwei — CHANGES_REQUESTED** [review]

**Comment by wilfred-s** [comment]
Just changing the entry to `google.protobuf.Any` does not fix the problem.
An `Any` is a  protobuf message which is serialised. It can be any message but still should to be a message. What are we going to store? What is the definition of the message. This is what it says in the [documentation](https://pkg.go.dev/google.golang.org/protobuf@v1.27.1/types/known/anypb):
> The Any message is a dynamic representation of any other message value. It is functionally a tuple of the full name of the remote message type and the serialized bytes of the remote message value.

Are you proposing to define a non protobuf based serialised version of the map then I do not think that this is the right approach.
I need a lot more detail on the how and what before we go ahead:

-1 until there is more detail on how this is going to be used

**Comment by chenya-zhang** [comment]
Discussed with Weiwei offline in https://yunikornworkspace.slack.com/archives/CL9CRJ1KM/p1632156003039900, we will close this PR and implement it using another approach in YuniKorn K8shim.

**Pull request closed without merging**